plugin.video.embycon/resources/lib/datamanager.py

```diff
@@ -0,0 +1,159 @@
+
+log = SimpleLogging(__name__)
+
+
+class DataManager():
+    cacheDataResult = None
+    dataUrl = None
+    cacheDataPath = None
+    canRefreshNow = False
+
+    def __init__(self, *args):
+        log.debug("DataManager __init__")
+
+    def getCacheValidatorFromData(self, result):
+        key = 'Items'
+        results = result.get(key)
+        if results is None:
+            key = 'SearchHints'
+            results = result.get(key)
+            if results is None:
+                results = []
+
+        itemCount = 0
+        dataHashString = ""
+
+        for item in results:
+            item_hash_string = getChecksum(item)
+            item_hash_string = str(itemCount) + "_" + key + "_" + item.get("Name", "-") + "_" + item_hash_string + "|"
+            log.debug("ITEM_HASH: " + item_hash_string)
+            dataHashString += item_hash_string
+            itemCount += 1
+
+        # hash the data
+        dataHashString = dataHashString.encode("UTF-8")
+        m = hashlib.md5()
+        m.update(dataHashString)
+        validatorString = m.hexdigest()
+
+        log.debug("getCacheValidatorFromData : RawData  : " + dataHashString)
+        log.debug("getCacheValidatorFromData : hashData : " + validatorString)
+
+        return validatorString
+
+    def loadJasonData(self, jsonData):
+        return json.loads(jsonData)
+
+    def GetContent(self, url):
+
+        __addon__ = xbmcaddon.Addon(id='plugin.video.embycon')
+        use_cache_system = __addon__.getSetting('cacheEmbyData') == "true"
+
+        if use_cache_system == False:
+            # dont use cache system at all, just get the result and return
+            log.debug("GetContent - Not using cache system")
+            jsonData = DownloadUtils().downloadUrl(url, suppress=False, popup=1)
+            result = self.loadJasonData(jsonData)
+            log.debug("Returning Loaded Result")
+            return result
+
+        # first get the url hash
+        m = hashlib.md5()
+        m.update(url)
+        urlHash = m.hexdigest()
+
+        # build cache data path
+
+        __addondir__ = xbmc.translatePath(__addon__.getAddonInfo('profile'))
+        if not os.path.exists(os.path.join(__addondir__, "cache")):
+            os.makedirs(os.path.join(__addondir__, "cache"))
+        cacheDataPath = os.path.join(__addondir__, "cache", urlHash)
+
+        log.debug("Cache_Data_Manager:" + cacheDataPath)
+
+        # are we forcing a reload
+        WINDOW = HomeWindow()
+        force_data_reload = WINDOW.getProperty("force_data_reload") == "true"
+        WINDOW.clearProperty("force_data_reload")
+
+        if os.path.exists(cacheDataPath) and not force_data_reload:
+            # load data from cache if it is available and trigger a background
+            # verification process to test cache validity   
+            log.debug("Loading Cached File")
+            with open(cacheDataPath, 'r') as f:
+                result = self.loadJasonData(f.read())
```

Comment:
you could just do
`self.cacheDataResult = self.loadJasonData(f.read())`
and remove
`self.cacheDataResult = result`

but that's superb minor and none relevant for this PR :)